internal/pkg/gateway/commitadapter.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package gateway
+
+import (
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/core/peer"
+	"github.com/pkg/errors"
+)
+
+type PeerNotifierAdapter struct {
+	Peer *peer.Peer
+}
+
+func (adapter *PeerNotifierAdapter) CommitNotifications(done <-chan struct{}, channelName string) (<-chan *ledger.CommitNotification, error) {
+	channel := adapter.Peer.Channel(channelName)
+	if channel == nil {
+		return nil, errors.Errorf("Channel does not exist: %s", channelName)
```

Comment:
From the material we reference in our coding [guidelines](https://github.com/hyperledger/fabric/blob/master/docs/source/style-guides/go-style.rst#coding-in-go):

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

As with most things Fabric there are plenty of counter-examples in legacy code but this is new code.

Reply:
I know that rule. My bad, this slipped past my radar

Reply:
Any idea why the staticcheck rules are not catching this?
https://staticcheck.io/docs/checks#ST1005
Is it because we prefer not to use standard Go error libraries, or are some rules disabled to avoid falling over on existing code (in which case, can we ramp up the rules applied to the Gateway code), or something else...?

Reply:
We disable the checks because the older code is littered with bad error strings.

https://github.com/hyperledger/fabric/blob/13092bbfb631a612091d4ccd3e849b320fa9c6a6/staticcheck.conf#L16-L19